src/types/Locking.ts

```diff
@@ -6,6 +6,7 @@ export default interface Lock {
   key: string;
   type: LockType;
   timestamp: Date;
+  expirationDate?: Date;
```

Comment:
Like I said in Serge's PR stacked over yours: relying on unix timestamp here will make things simple and speed.

Reply:
Already answered in my PR